src/engine/plugins/match.cpp

```diff
@@ -123,6 +124,8 @@ Status MatchPlugin::HandleRequest(const std::shared_ptr<datafacade::BaseDataFaca
         return Error("InvalidValue", "Invalid coordinate value.", json_result);
     }
 
+    // TODO: check is_sorted(ts, greater_equal), throw if not
```

Comment:
Done separately in https://github.com/Project-OSRM/osrm-backend/pull/3152 - should be merged into master asap.